hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/DBCheckpointServlet.java

```diff
@@ -172,7 +172,12 @@ private void generateSnapshotCheckpoint(HttpServletRequest request,
     }
 
     try (BootstrapStateHandler.Lock lock = getBootstrapStateLock().lock()) {
+      if (dbStore.getRocksDBCheckpointDiffer() != null) {
```

Comment:
synchronized ??

Reply:
It is not needed because count is atomic.
The reason I used `synchronized` here: https://github.com/apache/ozone/pull/5070/files#diff-8f8a9ef7fcf15970b94259f3d8dbe5a3d553dc8d10f0bce7cddebf28ada08b4cR224 because `notifyAll()` has to be called from `synchronized` block.